### PR TITLE
helics: add new package

### PR DESCRIPTION
**Pull request by nightlark**
A library/framework for synchronizing time and data between different simulators.

### CODE REVIEW

File: var/spack/repos/builtin/packages/helics/package.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+# them, you can save this file and test your package like this:
+#
+#     spack install helics
+#
+# You can edit this file again by typing:
+#
+#     spack edit helics
+#
+# See the Spack documentation for more information on packaging.
+# ----------------------------------------------------------------------------
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/helics/package.py]: You can delete everything below the license

File: var/spack/repos/builtin/packages/helics/package.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    variant('inproc', default=True, description="Enable in-process core type")
+    variant('mpi', default=False, description="Enable MPI core type")
+    variant('boost', default=True, description="Compile with Boost libraries")
+    variant('asio', default=True, description="Compile with ASIO libraries")
+    variant('swig', default=False, description="Build language bindings with SWIG")
+    variant('webserver', default=True, description="Enable the integrated webserver in the HELICS broker server")
+
+    # Build dependency
+    for _version in ['@master', '@develop']:
+        depends_on('git', type='build', when=_version)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/helics/package.py]: You can instead do:
```python
depends_on('git', type='build', when='@master:')
```
The open version range means "master or higher". `@master,develop` is also valid.

Comment (nightlark) [reply var/spack/repos/builtin/packages/helics/package.py]: Okay, so other branches (e.g. "test-branch") would be considered higher than master (and develop)?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/helics/package.py]: Nope, see #1983. There is special handling for develop, master, head, and trunk. Any other non-numeric version is sorted as being older than all other numeric versions.

File: var/spack/repos/builtin/packages/helics/package.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/helics/package.py]: Flake8:
```
var/spack/repos/builtin/packages/helics/package.py:8: [E302] expected 2 blank lines, found 1
```

File: var/spack/repos/builtin/packages/helics/package.py (hunk 2 above)
Comment (nightlark) [reply var/spack/repos/builtin/packages/helics/package.py]: Okay, got it.

**Pull request merged**